tests/app/notifications/rest/test_send_notification.py

```diff
@@ -572,128 +572,124 @@ def test_should_not_send_sms_if_team_api_key_and_not_a_service_user(notify_api,
         ] == json_resp['message']['to']
 
 
-def test_should_send_email_if_team_api_key_and_a_service_user(notify_api, sample_email_template, fake_uuid, mocker):
-    with notify_api.test_request_context(), notify_api.test_client() as client:
-        mocker.patch('app.celery.provider_tasks.deliver_email.apply_async')
-        mocker.patch('app.dao.notifications_dao.create_uuid', return_value=fake_uuid)
+def test_should_send_email_if_team_api_key_and_a_service_user(client, sample_email_template, fake_uuid, mocker):
+    mocker.patch('app.celery.provider_tasks.deliver_email.apply_async')
+    mocker.patch('app.notifications.process_notifications.uuid.uuid4', return_value=fake_uuid)
 
-        data = {
-            'to': sample_email_template.service.created_by.email_address,
-            'template': sample_email_template.id
-        }
-        api_key = ApiKey(service=sample_email_template.service,
-                         name='team_key',
-                         created_by=sample_email_template.created_by,
-                         key_type=KEY_TYPE_TEAM)
-        save_model_api_key(api_key)
-        auth_header = create_jwt_token(secret=api_key.secret, client_id=str(api_key.service_id))
+    data = {
+        'to': sample_email_template.service.created_by.email_address,
+        'template': sample_email_template.id
+    }
+    api_key = ApiKey(service=sample_email_template.service,
+                     name='team_key',
+                     created_by=sample_email_template.created_by,
+                     key_type=KEY_TYPE_TEAM)
+    save_model_api_key(api_key)
+    auth_header = create_jwt_token(secret=api_key.secret, client_id=str(api_key.service_id))
```

Comment:
turns out the `create_authorization_header` creates api keys of the appropriate type if they don't exist, so yes, good thinking